Add additional information to note for referencing packed field diagnostic

bindsdev opened the pull request:
Addresses #110199

chenyukang committed:
configure --set support list as arguments

chenyukang committed:
add test

bindsdev committed:
add additional information to note for referencing packed field diagnostic

TaKO8Ki commented:
Could you fix some unit test stderr files?

bindsdev committed:
fix failing tests

chenyukang committed:
Fix help message of option for checking List

Noratrieb committed:
Document `const {}` syntax for `std::thread_local`.

It exists and is pretty cool. More people should use it.

m-ou-se committed:
Remove "V1" from ArgumentsV1 and FlagsV1.

m-ou-se committed:
Update tests.

Commit:
test that we format the panic message only once

m-ou-se committed:
Restructure std::fmt::rt a bit.

This moves more of the internal/lang items into the private rt module.

m-ou-se committed:
Update miri test.

WaffleLapkin committed:
Fix `ui/io-checks/inaccessbile-temp-dir.rs` test

WaffleLapkin committed:
Improve `non-ice-error-on-worker-io-fail.rs` test

- Remove unrelated code
- Use a local dir path

scottmcm committed:
Add some MIR pre-codegen tests for slice indexing

scottmcm committed:
Also use `mir::Offset` for pointer `add`

chenyukang committed:
fix bug in set args

compiler-errors commented on compiler/rustc_mir_transform/src/check_packed_ref.rs:
```diff
@@ -56,8 +56,9 @@ impl<'tcx> Visitor<'tcx> for PackedRefChecker<'_, 'tcx> {
                         "reference to packed field is unaligned"
                     )
                     .note(
```
This should probably be broken up into several notes, e.g.

> packed structs are only aligned by one byte, and many modern architectures penalize unaligned field accesses

> creating a misaligned reference is undefined behavior (even if that ...

compiler-errors replied:
I tried to phrase my suggestion to avoid these periods in the note, because they don't match the style of other notes. Can you rephrase this to be more like what I suggested here?

bindsdev replied:
Oh yeah, I totally missed that one. Apologies.

clubby789 committed:
Migrate trivially translatable `rustc_parse` diagnostics

bvanjoi committed:
refactor(docs): remove macro resolution fallback

bvanjoi committed:
test(doc): no fallback marco resolution

BoxyUwU committed:
nits

WaffleLapkin committed:
fix typo in the `inaccessible-temp-dir` test name

(formally `...inaccessbile...`)

WaffleLapkin committed:
ignore failing test on arm

m-ou-se committed:
Remove unused std::sys_common::thread_local_key::Key.

m-ou-se committed:
Update test.

GuillaumeGomez committed:
repr attribute needs to be stored to be used in doc(inline) by rustdoc

GuillaumeGomez committed:
Add regression test for #110698

m-ou-se committed:
pub -> pub(super).

fmease committed:
rustdoc: rebind bound vars to type-outlives predicates

john-h-k committed:
Skip rustc version detection on macOS

compiler-errors committed:
Add invalid_macro_export_arguments to built-in macro list

compiler-errors committed:
Sort hardwired lints

compiler-errors committed:
Add a few more missing lints

john-h-k committed:
Comment round #1

ChrisDenton committed:
remove_dir_all: delete directory with fewer perms

If opening a directory with `FILE_LIST_DIRECTORY` access fails then we should try opening without requesting that access. We may still be able to delete it if it's empty or a link.

cuishuang committed:
Fix unavailable url

Signed-off-by: cui fliter <imcusg@gmail.com>

bindsdev committed:
Merge branch 'master' into improve-packed-struct-ref-diagnostic

bindsdev committed:
split note into two

compiler-errors reviewed (changes requested):
Please see my follow-up comment about the phrasing of the note message. Also, Merge commits are not allowed in PRs, and this really should be squashed into one commit.

ehuss committed:
Update awscli

bindsdev commented:
I messed something up. I am going to reopen the PR.
